src/wp-includes/class-wp-block.php

```diff
@@ -187,19 +253,74 @@ public function __get( $name ) {
 	 * @param array $options {
 	 *   Optional options object.
 	 *
-	 *   @type bool $dynamic Defaults to 'true'. Optionally set to false to avoid using the block's render_callback.
+	 *   @type bool $dynamic     Defaults to true. Optionally set to false to avoid using the block's `render_callback`.
+	 *   @type bool $filter_data Defaults to false. When true, `$parsed_block` and `$available_context` are filtered using `render_block_data` and `render_block_context` and the block's derived properties are set again.
 	 * }
 	 * @return string Rendered block output.
 	 */
 	public function render( $options = array() ) {
 		global $post;
+
+		/** This filter is documented in wp-includes/blocks.php */
+		$pre_render = apply_filters( 'pre_render_block', null, $this->parsed_block );
```

Comment:
I notice this is still defined in `render_block` and not removed like render_block_data and render_block_context.

Reply:
My intention was that calling it twice would let one short-circuit `render_block` **and** `WP_Block::render`. Now that I think about it, though, maybe that will cause unexpected results for folks using `pre_render_block` for things other than `return false;`.